torch/_functorch/_aot_autograd/input_output_analysis.py

```diff
@@ -49,12 +49,16 @@ def remove_dupe_metadata(
     other_traced_tangents = m.traced_tangents[num_data_mutations:]
     inp_traced_tangents = m.traced_tangents[:num_data_mutations]
     filtered_inp_traced_tangents = [
-        # See Note [Tangents must be contiguous]
+        # See Note [Tangents memory format]
         x
         for i, x in enumerate(inp_traced_tangents)
         if keep_arg_mask[m.mutated_inp_runtime_indices[i]]
     ]
     traced_tangents = filtered_inp_traced_tangents + other_traced_tangents
+    assert m.traced_tangent_memory_formats is not None
```

Comment:
side note (since I see the logic you're changing here is in the `remove_dupe_metadata` codepath): we are pretty sure that this is dead code in torch.compile, since dynamo has its own logic to remove dupe'd inputs.

@jamesjwu tried to kill this code a while back (but I think ran into some issues?) - we probably want to give it another shot at some point https://github.com/pytorch/pytorch/pull/127306